src/string/startCase.ts

```diff
@@ -0,0 +1,30 @@
+import { getWords } from './_internal/getWords';
+
+/**
+ * Converts the first character of each word in a string to uppercase and the remaining characters to lowercase.
+ *
+ * Start case is the naming convention in which each word is written with an initial capital letter.
+ * @param {string} str - The string to convert.
+ * @returns {string} The converted string.
+ *
+ * @example
+ * const result1 = startCase('hello world');  // result will be 'Hello World'
+ * const result2 = startCase('HELLO WORLD');  // result will be 'Hello World'
+ * const result3 = startCase('hello-world');  // result will be 'Hello World'
+ * const result4 = startCase('hello_world');  // result will be 'Hello World'
+ */
+export function startCase(str: string): string {
+  const words = getWords(str.trim());
+  let result = '';
+  for (const word of words) {
+    if (result) {
+      result += ' ';
+    }
+    if (word === word.toUpperCase()) {
+      result += word;
+    } else {
+      result += word[0].toUpperCase() + word.slice(1).toLowerCase();
+    }
+  }
```

Comment:
```suggestion
  return words.map(word => {
    if (word === word.toUpperCase()) {
      return word;
    } else {
      return word[0].toUpperCase() + word.slice(1).toLowerCase();
    }
  }).join(' ');
```

I thought this implementation is more readable. Is using for loops because they are more performant? If so, how much is it more performant?

Reply:
![image](https://github.com/user-attachments/assets/13e33198-8ab9-4582-9a40-1b0ae3ef8a10)

Using the `for of` loop in the `startCase` function is faster than using the `map` function.
The image above compares `map[startCase]` and `for of[startCase]` using benchmark tests.